### PR TITLE
fix rls in a rustup environment where every folder can be a different rustc (e.g. stable)

### LINKED ISSUE #25: TypeError: Cannot read property 'dispose' of null
I tried to follow the setup instructions. I also set `rustup override set nightly` on both the `rls` directory and my Rust project directory.

vcscode 1.8.1, and all RLS checkouts from just a few minutes ago.

(One oddity about my system: The language is set to French, and vscode gives me some kind of cut off warning messages which might be related to missing messages. I tried re-running it English mode, but it the debugged copy of vscode doesn't honor environment variables set in the shell which launched the host vscode.)

```
/usr/share/code/code --debugBrkPluginHost=6242 --extensionDevelopmentPath=/home/emk/w/src/rust-libs/rls_vscode 
TypeError: Cannot read property 'dispose' of null
    at LanguageClient.cleanUp (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-languageclient/lib/main.js:647:30)
    at LanguageClient.stop (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-languageclient/lib/main.js:621:14)
    at LanguageClient.handleConnectionError (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-languageclient/lib/main.js:882:18)
    at errorHandler (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-languageclient/lib/main.js:744:19)
    at /home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-languageclient/lib/main.js:47:42
    at CallbackList.invoke (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-jsonrpc/lib/events.js:103:39)
    at Emitter.fire (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-jsonrpc/lib/events.js:167:36)
    at writeErrorHandler (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-jsonrpc/lib/main.js:113:22)
    at CallbackList.invoke (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-jsonrpc/lib/events.js:103:39)
    at Emitter.fire (/home/emk/w/src/rust-libs/rls_vscode/node_modules/vscode-jsonrpc/lib/events.js:167:36)
```

### PR DESCRIPTION
**Pull request by oli-obk**
addresses the `rustup` aspect of #25

diff is probably best viewed with whitespace ignored: https://github.com/jonathandturner/rls_vscode/pull/31/files?w=1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,3 +32,26 @@ Mac/Linux: $HOME/.vscode/extensions
 ```
 For example, to setup a symlink on Mac/Linux: `ln -s /path/to/rls_vscode/ ~/.vscode/extensions/rls_vscode`
 Restart VSCode in order to load the extension. More information available via [VSCode docs](https://code.visualstudio.com/Docs/extensions/example-hello-world#_installing-your-extension-locally).
+
+## Troubleshooting
+
+### something .. something ... `tsc -watch -p ./` or `ENOSPC`
```
Comment (jonathandturner) [inline README.md]: something something?

**Comment by jonathandturner** [comment]
Looking over this (and, admittedly, it's in the morning here so my brain may not be awake), but how does this address #25?  It seems to be a cleanup of the existing logic?

File: README.md (hunk 1 above)
Comment (oli-obk) [inline README.md]: well... the text around that is magical and incomprehensible to the unskilled javascript mind (like mine)

**Comment by oli-obk** [comment]
> but how does this address #25?

it changes the calls from using `--manifest-path` to actually changing the current directory to the directory of `rls`. This allows use to use `rustup override set nightly` on the `rls` dir, but work on a stable project.

It does not fix the error message in #25 in general. If you forget to set the override or the build fails out of other reasons, you still get that error message. So indeed, this is not a real fix, but fixes one of the issues leading to that error message. I removed the `fixes` comment in the OP.

File: README.md (hunk 1 above)
Comment (jonathandturner) [inline README.md]: Surely we don't want the readme to say "something .. something .."

**Comment by oli-obk** [comment]
removed "something ... something ..."

**Pull request merged**